### PR TITLE
Use json2bin to build pl_personal.narc

### PR DESCRIPTION
**Pull request by wildfire248**
Uses json2bin to build pl_personal.narc from source files.
Also made some edits to the json2bin tool:
- Now uses the `@PRIVATE_DIR@` argument from meson for intermediate files (see `custom_target()` in meson docs)
- Added support for using list indexes in schema paths, e.g. `"types.0"` in schema gets `obj["types"][0]`, where `obj` is the loaded json file. (Decimal numbers only)

### New Files
- `res/pokemon/`
  - `[folder per pokemon]/`
    - `forms/` (only if pokemon has multiple forms with different base stats)
      - `[folder per form]/`
        - `data.json`
    - `data.json`
    - `meson.build`
  - `meson.build`
- `tools/json2bin/`
  - `consts/`
    - `item.py`
    - `pokemon/`
      - `ability.py`
      - `color.py`
      - `egg_group.py`
      - `exp_rate.py`
      - `gender_ratio.py`
      - `species.py`
      - `tms.py`
  - `meson.build`
  - `pokemon_personal_data.py`

### Renamed Files
- `tools/json2bin/consts/poketype.py` -> `tools/json2bin/consts/pokemon/mon_type.py`

### Edited Files
- `include/`
  - `constants/pokemon.h`
  - `pokemon.h`
- `platinum.us/rom.rsf`
- `res/`
  - `battle/moves/meson.build`
  - `prebuilt/poketool/personal/meson.build`
  - `meson.build`
- `src/pokemon.c`
- `tools/json2bin/`
  - `json2bin.py`
  - `movedata.py`

### Deleted Files
- `res/prebuilt/poketool/personal/pl_personal.narc`

### CODE REVIEW

File: src/pokemon.c [hunk 1]
```diff
@@ -1946,8 +1946,8 @@ u32 PokemonPersonalData_GetValue(PokemonPersonalData *monPersonalData, enum Poke
     case MON_DATA_PERSONAL_COLOR:
         result = monPersonalData->color;
         break;
-    case MON_DATA_PERSONAL_INVERSE:
-        result = monPersonalData->inverse;
+    case MON_DATA_PERSONAL_FLIP_SPRITE:
+        result = monPersonalData->flip_sprite;
```
Comment (lhearachel) [inline src/pokemon.c:1950]: Camel case

File: res/pokemon/meson.build [hunk 2]
```diff
@@ -0,0 +1,534 @@
...
+subdir('regigigas')
+subdir('giratina')
+subdir('cresselia')
+subdir('phione')
+subdir('manaphy')
+subdir('darkrai')
+subdir('shaymin')
+subdir('arceus')
+subdir('494')  # Pokemon Egg
+subdir('495')  # Bad Egg TODO verify this
```
Comment (lhearachel) [inline res/pokemon/meson.build:501]: From experience, it's the bad egg.

File: tools/json2bin/consts/item.py [hunk 3]
```diff
@@ -0,0 +1,472 @@
...
+    ITEM_LUCK_INCENSE = auto()
+    ITEM_PURE_INCENSE = auto()
+    ITEM_PROTECTOR = auto()
+    ITEM_ELECTIRIZER = auto()
+    ITEM_MAGMARIZER = auto()
+    ITEM_DUBIOUS_DISC = auto()
+    ITEM_REAPER_CLOTH = auto()
+    ITEM_RAZOR_CLAW = auto()
+    ITEM_RAZOR_FANG = auto()
+    ITEM_TM01 = auto()
```
Comment (lhearachel) [inline tools/json2bin/consts/item.py:333]: Do we want to pad these numbers to 3 digits for future modders?

File: tools/json2bin/pokemon_personal_data.py [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+    .register('egg_groups.1', 1, j2b.parse_const, egg_group.PokemonEggGroup) \
+    .register('abilities.0', 1, j2b.parse_const, ability.PokemonAbility) \
+    .register('abilities.1', 1, j2b.parse_const, ability.PokemonAbility) \
+    .register('great_marsh_flee_rate', 1, j2b.parse_int) \
+    .register('sprite', 1, parse_color) \
+    .pad(2) \
+    .register('learnset.tms', 16, j2b.pack_flags, tms.PokemonLearnsetTMs)
+
+
+FORM_INDEXES = {
```
Comment (lhearachel) [inline tools/json2bin/pokemon_personal_data.py:65]: `FORM_INDICES`

**Review by lhearachel — CHANGES_REQUESTED** [review]

File: res/pokemon/meson.build (hunk 2 above)
Comment (Nomura-RH) [reply res/pokemon/meson.build:501]: Is there are a reason we can't just name those folders egg and bad_egg?

**Pull request merged**